ManageIQ/manageiq
# Add lib/workers/bin/run_single_worker.rb script

Allows running of workers via a single script.

Currently skips removing db records, so stale MiqWorker records will still be left in the DB.  Also, doesn't load the "MVP" number of classes, which will be done in a separate effort, but allows for any worker known to by MIQ to be spawned.

Requires disabling heartbeating, unless a `MiqServer::WorkerManagement` process is running (disabling heartbeating is done in https://github.com/ManageIQ/manageiq/pull/15123 ).

Links
-----
* https://www.pivotaltracker.com/story/show/145276189
* https://github.com/ManageIQ/manageiq/pull/15123


Steps for Testing/QA
--------------------
~~If https://github.com/ManageIQ/manageiq/pull/15123 is not merged yet, run the following in this branch:~~  **UPDATE**:  #15123 is now merged into this branch, so the following is now not needed.

```console
$ git apply <(curl -L https://github.com/ManageIQ/manageiq/pull/15123.patch)
```

From there you can try out the following:

```console
$ bundle exec ruby lib/workers/bin/run_single_worker.rb
$ bundle exec ruby lib/workers/bin/run_single_worker.rb -h
$ bundle exec ruby lib/workers/bin/run_single_worker.rb -l
$ bundle exec ruby lib/workers/bin/run_single_worker.rb MiqUiWorker
$ bundle exec ruby lib/workers/bin/run_single_worker.rb [PICK_CLASS_FROM_DASH_L_LIST_FROM_ABOVE]
```

And confirm that the last to start the worker as expected.

Actually, this now doesn't require `bundle exec` to function, and you can clearly see the difference now by running it with and without using the `-h` or `-l` flags.  Even passing in something like `FakeWorker` as an argument will be caught much quicker thanks to the last two commits.
